src/Microsoft.TemplateEngine.Authoring.TemplateVerifier/VerificationEngine.cs

```diff
@@ -0,0 +1,335 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.TemplateEngine.Authoring.TemplateVerifier.Commands;
+using Microsoft.TemplateEngine.Utils;
+using Xunit.Abstractions;
+
+namespace Microsoft.TemplateEngine.Authoring.TemplateVerifier
+{
+    public class VerificationEngine
+    {
+        private static readonly IReadOnlyList<string> _defaultVerificationExcludePatterns = new List<string>()
+        {
+            @"obj/*",
+            @"obj\*",
+            @"bin/*",
+            @"bin\*",
+            "*.exe",
+            "*.dll",
+            "*.",
+            "*.exe",
+        };
+
+        private readonly TemplateVerifierOptions _options;
+        private readonly ILogger _logger;
+
+        public VerificationEngine(IOptions<TemplateVerifierOptions> optionsAccessor, ILogger logger)
+        {
+            if (optionsAccessor == null)
+            {
+                throw new ArgumentNullException(nameof(optionsAccessor));
+            }
+
+            _options = optionsAccessor.Value;
+            _logger = logger;
+        }
+
+        public async Task Execute(CancellationToken cancellationToken = default)
+        {
+            //TODO: add functionality for uninstalled templates from a local folder
+            if (!string.IsNullOrEmpty(_options.TemplatePath))
+            {
+                throw new TemplateVerificationException("Custom template path not yet supported.", TemplateVerificationErrorCode.InternalError);
```

Comment:
```suggestion
                throw new TemplateVerificationException("Custom template path not yet supported.", TemplateVerificationErrorCode.NotSupported);
``` 
just an idea

Reply:
Good suggestion.
This limitation is currently planned to be addressed as MVP (I see that as must-have) - so leaving the error as is now - it's just a placeholder.